Lesson15/ecenaz/script.js

```diff
@@ -0,0 +1,630 @@
+*/
+function checkProtocol(url) {
+  const lowerCaseUrl = url.toLowerCase();
+
+  if (lowerCaseUrl.startsWith("https")) {
+    console.log("Secure connection");
+  } else {
+    console.log("Unsecure connection");
+  }
+}
+checkProtocol("https://www.example.com");
+checkProtocol("HELLO://www.example.com");
+
+
+/*
+12. Switch: Day of the Week
+   - Define a function `getDayOfWeek(num)` that uses a switch statement:
+     1 -> "Monday"
+     2 -> "Tuesday"
+     ...
+     7 -> "Sunday"
+     - Log the matched day or "Invalid day" if out of range.
+*/
+
+function getDayOfWeek(num) {
+  switch (num) {
+    case 1:
+      console.log("Monday");
+      break;
+    case 2:
+      console.log("Tuesday");
+      break;
+    case 3:
+      console.log("Weddnesday");
+      break;
+    case 4:
+      console.log("Thursday");
+      break;
+    case 5:
+      console.log("Friday");
+      break;
+    case 6:
+      console.log("Saturday");
+      break;
+    case 7:
+      console.log("Sunday");
+      break;
+    default:
+      console.log("Unknow day");
+      break;
+  }
+}
+
+getDayOfWeek(5);
+getDayOfWeek(4);
+getDayOfWeek(1);
+getDayOfWeek('hello');
+getDayOfWeek(10);
+
+
+
+/*
+13. Repeat a String
+   - Define a function `repeatWord(word, times)` that uses the .repeat() method
+     to repeat `word` `times` times.
+   - Log the repeated result.
+   
+*/
+function repeatWord(word, times) {
+
+  const repeatedWord = word.repeat(times);
+
+
+  console.log(repeatedWord);
+}
+repeatWord("hello world", 3);
+
+/*
+14. Replace Substring
+   - Define a function `censorWord(sentence, target)` that replaces `target`
+     with "****" (use .replaceAll() or multiple .replace()).
+   - Log the censored sentence.
+*/
+function censorWord(sentence, target) {
+  const censoredWord = sentence.replaceAll(target, "*****");
+
+  console.log(censoredWord);
+}
+censorWord("This is a bad sentence with bad words.", "bad");
+
+/*
+15. Check First Character (if-else)
+   - Define a function `startsWithA(str)` that checks if the string starts with 'A'
+     (use .charAt(0) or [0]).
+   - Log "Starts with A" or "Does not start with A".
+*/
+function startWithA(str) {
+
+  if (str.charAt(0) === 'A') {
+    console.log("starts With A");
+  } else {
+    console.log("Does not start with A");
+  }
+
+}
+
+startWithA("Apple");
+startWithA("Banana");
+/*
+16. Slice Last N Characters
+   - Define a function `sliceLastN(text, n)` that uses .slice(-n) to extract
+     the last `n` characters of `text`.
+   - Log the result.
+*/
+function sliceLastN(text, n) {
+  const slicedText = text.slice(-n);
+
+  console.log(slicedText);
+}
+sliceLastN("Hello, World!", 5);
+/*
+17. Switch: Grade Checker
+   - Define a function `gradeChecker(score)` that uses a switch (or if-else chain):
+     90+ -> "A"
+     80-89 -> "B"
+     70-79 -> "C"
+     60-69 -> "D"
+     below 60 -> "F"
+   - Log the grade.
+*/
+
+
+
+function gradeChecker(score) {
+
+  switch (score) {
```

Comment:
In JS, the switch statement checks the exact match. E.g. 'score === '90+''.
Therefore, the solution here would not work. 

Instead, it's better to use if/else statement here. Conditions would be like in math:
`if (score > 90)`